pkg/azdiskschedulerextender/azdiskschedulerextender.go

```diff
@@ -24,42 +24,57 @@ import (
 	"time"
 
 	v1 "k8s.io/api/core/v1"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
 	"k8s.io/client-go/tools/clientcmd"
 	"k8s.io/klog/v2"
 	schedulerapi "k8s.io/kube-scheduler/extender/v1"
-
 	v1alpha1Meta "sigs.k8s.io/azuredisk-csi-driver/pkg/apis/azuredisk/v1alpha1"
-	clientSet "sigs.k8s.io/azuredisk-csi-driver/pkg/apis/client/clientset/versioned/typed/azuredisk/v1alpha1"
+	"sigs.k8s.io/azuredisk-csi-driver/pkg/apis/client/clientset/versioned"
+	informers "sigs.k8s.io/azuredisk-csi-driver/pkg/apis/client/informers/externalversions"
+	azurediskInformers "sigs.k8s.io/azuredisk-csi-driver/pkg/apis/client/informers/externalversions/azuredisk/v1alpha1"
 )
 
 var (
-	azVolumeAttachmentExtensionClient clientSet.AzVolumeAttachmentInterface
-	azDriverNodeExtensionClient       clientSet.AzDriverNodeInterface
-	kubeExtensionClientset            *clientSet.DiskV1alpha1Client
-	ns                                = "azure-disk-csi" //TODO update to correct namespace when finalized
+	azVolumeAttachmentInformer azurediskInformers.AzVolumeAttachmentInformer
+	azDriverNodeInformer       azurediskInformers.AzDriverNodeInformer
+	kubeExtensionClientset     versioned.Interface
+	ns                         = "azure-disk-csi" //TODO update to correct namespace when finalized
 )
 
 type azDriverNodesMeta struct {
-	nodes *v1alpha1Meta.AzDriverNodeList
+	nodes []*v1alpha1Meta.AzDriverNode
```

Comment:
I totally agree with you. The only reason I had to do this is because `azDriverNodeInformer.Lister().AzDriverNodes(ns).List(labels.Everything())` and `azVolumeAttachmentInformer.Lister().AzVolumeAttachments(ns).List(labels.Everything())` return array of pointers. I assume this is because `AzDriverNode` and `AzVolumeAttachment` contains a lot of data and this is done to avoid making a copy.